specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/functions.json

```diff
@@ -0,0 +1,811 @@
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "ScalarFunctionProperties": {
+      "description": "The properties that are associated with a scalar function.",
+      "x-ms-discriminator-value": "Scalar",
+      "allOf": [
+        {
+          "$ref": "#/definitions/FunctionProperties"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ScalarFunctionConfiguration",
+          "description": "Describes the configuration of the scalar function.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ScalarFunctionConfiguration": {
+      "description": "Describes the configuration of the scalar function.",
+      "properties": {
+        "inputs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FunctionInput"
+          },
+          "description": "A list of inputs describing the parameters of the function."
+        },
+        "output": {
+          "$ref": "#/definitions/FunctionOutput",
+          "description": "The output of the function."
+        },
+        "binding": {
+          "$ref": "#/definitions/FunctionBinding",
+          "description": "The physical binding of the function. For example, in the Azure Machine Learning web service’s case, this describes the endpoint."
+        }
+      }
+    },
+    "FunctionInput": {
+      "description": "Describes one input parameter of a function.",
+      "properties": {
+        "dataType": {
+          "type": "string",
+          "description": "The (Azure Stream Analytics supported) data type of the function input parameter. A list of valid Azure Stream Analytics data types are described at https://msdn.microsoft.com/en-us/library/azure/dn835065.aspx"
+        },
+        "isConfigurationParameter": {
+          "type": "boolean",
+          "description": "A flag indicating if the parameter is a configuration parameter. True if this input parameter is expected to be a constant. Default is false."
+        }
+      }
+    },
+    "FunctionOutput": {
+      "description": "Describes the output of a function.",
+      "properties": {
+        "dataType": {
+          "type": "string",
+          "description": "The (Azure Stream Analytics supported) data type of the function output. A list of valid Azure Stream Analytics data types are described at https://msdn.microsoft.com/en-us/library/azure/dn835065.aspx"
+        }
+      }
+    },
+    "FunctionBinding": {
+      "description": "The physical binding of the function. For example, in the Azure Machine Learning web service’s case, this describes the endpoint.",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Indicates the function binding type."
+        }
+      },
+      "required": [
```

Comment:
>required [](http://example.com/codeflow?start=7&length=8)

Changing a property to required is a new breaking change. Was it covered by your breaking change review/approval?